### PR TITLE
Update CreateChangeRequestView.tsx

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

Added functionality for the CR create page to sort its dropdown for projects by wbs number

## Screenshots
Screen Shot 2023-09-11 at 9 46 59 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.

Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #536)

### CODE REVIEW

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 1]
```diff
@@ -105,6 +105,21 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
...
+    if (wbsNumber1 < wbsNumber2) {
+      return -1;
+    } else if (wbsNumber1 > wbsNumber2) {
+      return 1;
+    } else {
+      return 0;
+    }
+  };
+
   projects.forEach((project: Project) => {
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:123]: Actually @RChandler234 dont we have a function for this somewhere

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 2]
```diff
@@ -105,6 +105,21 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
   const projectOptions: { label: string; id: string }[] = [];
   const wbsDropdownOptions: { label: string; id: string }[] = [];
 
+  const customComparator = (option1: { id: string }, option2: { id: string }) => {
+    // Extract the WBS numbers from the options and convert them to a sortable format
+    const wbsNumber1 = parseFloat(option1.id.replace(/-/g, '.'));
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:110]: i dont think this is explicitly seperating projects and work packages but it seems to be working but idk 🤷‍♂️

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
looks awesome nice job harish

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx (hunk 1 above)
Comment (RChandler234) [reply src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:123]: I don't think so, it appears the tables just happen to do it

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 3]
```diff
@@ -105,6 +105,21 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
   const projectOptions: { label: string; id: string }[] = [];
   const wbsDropdownOptions: { label: string; id: string }[] = [];
 
+  const customComparator = (option1: { id: string }, option2: { id: string }) => {
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:108]: this method is a bit overfitted to the use-case, I think we could actually end up using this in a lot of different places. Rename it to wbsNumComparator and have it in two strings, wbsNum1, wbsNum2; Also move it to src\shared\src\validate-wbs.ts

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 4]
```diff
@@ -105,6 +105,21 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
...
+    const wbsNumber2 = parseFloat(option2.id.replace(/-/g, '.'));
+
+    // Compare the WBS numbers and return -1, 0, or 1 based on the comparison
+    if (wbsNumber1 < wbsNumber2) {
+      return -1;
+    } else if (wbsNumber1 > wbsNumber2) {
+      return 1;
+    } else {
+      return 0;
+    }
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:120]: this logic may technically work, but we should compare the elements of a wbsNum (car number, project number, and work package number) individually, to make sure we're being more precise. You can do this by calling .split('.') on each wbsNum string to get an array of strings [carNumber, projectNumber, wpNumber]. You can then compare the carNumber1 to carNumber2 and return -1 or 1 if one is greater/less or go on to check the projectNumber if they're equal (and so on)

Comment (RChandler234) [reply src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:120]: lmk if that's confusing

**Review by RChandler234 — CHANGES_REQUESTED** [review]
remember to tag your next commit!

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx (hunk 4 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:120]: You could also just use validateWbs function to get an object in that format and compare like reid said so u dont have to mess with split strings

**Comment by RChandler234** [comment]
You need to tag your commits

**Pull request closed without merging**